src/types.jl

```diff
@@ -149,14 +150,30 @@ end
 For varargs method definitions since they generate multiple method objects. Use
 the *same* Entry object for each object's documentation.
 """
-function setmeta!(modname, objects::Set, category, source, args...)
+function setmeta!(modname, objects::Set, category, source, code, args...)
     entry = Entry{category}(modname, source, args...)
+    addentry_codedata!(entry, code)
     meta = getdoc(modname)
     for object in objects
         pushmeta!(meta, object, entry)
     end
 end
 
+"Adds and filters which code are added to entry.data"
+function addentry_codedata!(entry::Entry, code)
+    if (ismethod(code)  ||
+        ismacro(code)   ||
+        istype(code)    ||
+        isalias(code)   ||
+        isglobal(code)  ||
+        issymbol(code)  ||
+        isquote(code)
+        )
```

Comment:
I'd suggest only capturing `method` and `macro` for the moment. `alias`, `global`, `symbol`, and `quote` don't give us anything useful. Reasons for not capturing `type` I've mentioned in the main comments as a reply.